pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -0,0 +1,794 @@
+
+	return nil
+}
+
+func (h *Handler) checkValueIsChanged(setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	var updatedSetting *harvesterv1.Setting
+	var err error
+	if !h.checkIsSameHashValue(setting) {
+		if setting.Value == "" {
+			setting.Annotations[OldNadStorageNetworkAnnotation] = setting.Annotations[NadStorageNetworkAnnotation]
+			setting.Annotations[NadStorageNetworkAnnotation] = ""
+			h.setHashAnnotations(setting)
+
+			if updatedSetting, err = h.setConfiguredCondition(setting, false, ReasonInProgress, "enqueue"); err != nil {
+				return setting, fmt.Errorf("reset update status error %v", err)
+			}
+			return updatedSetting, errors.New("check hash again")
+		} else {
+			if err = h.createAndSaveNad(setting); err != nil {
+				return setting, err
+			}
+
+			h.setHashAnnotations(setting)
+
+			if updatedSetting, err = h.setConfiguredCondition(setting, false, ReasonInProgress, "create nad"); err != nil {
+				return setting, fmt.Errorf("create nad update status error %v", err)
+			}
+			return updatedSetting, errors.New("check hash again")
+		}
+	}
+	return setting, nil
+}
+
+func (h *Handler) removeOldNad(setting *harvesterv1.Setting) error {
+	oldNad := setting.Annotations[OldNadStorageNetworkAnnotation]
+	if oldNad == "" {
+		return nil
+	}
+
+	nadName := strings.Split(oldNad, "/")
+	if len(nadName) != 2 {
+		logrus.Errorf("split nad namespace and name failed %s", oldNad)
+		setting.Annotations[OldNadStorageNetworkAnnotation] = ""
+		return nil
+	}
+	namespace := nadName[0]
+	name := nadName[1]
+
+	if _, err := h.networkAttachmentDefinitionsCache.Get(namespace, name); err != nil {
+		if apierrors.IsNotFound(err) {
+			setting.Annotations[OldNadStorageNetworkAnnotation] = ""
+			return nil
+		}
+
+		// retry again
+		return fmt.Errorf("check net-attach-def existing error %v", err)
+	}
+
+	if err := h.networkAttachmentDefinitions.Delete(namespace, name, &metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("remove nad error %v", err)
+	}
+
+	setting.Annotations[OldNadStorageNetworkAnnotation] = ""
+	return nil
+}
+
+func (h *Handler) handleLonghornSettingPostConfig(setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	// check if we need to restart monitoring pods
+	if err := h.checkPodStatusAndStart(); err != nil {
+		if _, updateConditionErr := h.setConfiguredCondition(setting, false, ReasonInProgress, MsgRestartPod); updateConditionErr != nil {
+			return setting, fmt.Errorf("update status error %v", updateConditionErr)
+		}
+		return setting, err
+	}
+
+	settingCopy := setting.DeepCopy()
+	if err := h.removeOldNad(settingCopy); err != nil {
+		return setting, fmt.Errorf("remove old nad error %v", err)
+	}
+
+	var s *harvesterv1.Setting
+	var err error
+	if s, err = h.setConfiguredCondition(settingCopy, true, ReasonCompleted, ""); err != nil {
+		return setting, fmt.Errorf("update status error %v", err)
+	}
```

Comment:
```suggestion
	s, err := h.setConfiguredCondition(settingCopy, true, ReasonCompleted, ""); 
	if err != nil {
		return setting, fmt.Errorf("update status error %v", err)
	}
```
preferred for good project code consistency.

Reply:
fix